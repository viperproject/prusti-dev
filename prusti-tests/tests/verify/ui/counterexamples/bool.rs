// compile-flags: -Pcounterexample=true

use prusti_contracts::*;

#[ensures(result)]
fn test1(b: bool) -> bool {
    !b
}

#[pure]
#[ensures(result)]
fn test2(b: bool) -> bool {
    !b
}

fn test3(b: bool) {
    assert!(b);
}

fn main() {}
