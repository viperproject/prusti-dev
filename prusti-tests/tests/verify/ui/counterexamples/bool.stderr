error: [Prusti: verification error] postcondition might not hold.
 --> $DIR/bool.rs:5:11
  |
5 | #[ensures(result)]
  |           ^^^^^^
  |
note: the error originates here
 --> $DIR/bool.rs:6:1
  |
6 | / fn test1(b: bool) -> bool {
7 | |     !b
8 | | }
  | |_^
note: counterexample for "b"
  initial value: true
  final value:   true
 --> $DIR/bool.rs:6:10
  |
6 | fn test1(b: bool) -> bool {
  |          ^
note: counterexample for result
  final value:   false
 --> $DIR/bool.rs:6:22
  |
6 | fn test1(b: bool) -> bool {
  |                      ^^^^

error: [Prusti: verification error] postcondition might not hold.
  --> $DIR/bool.rs:11:11
   |
11 | #[ensures(result)]
   |           ^^^^^^
   |
note: the error originates here
  --> $DIR/bool.rs:12:1
   |
12 | / fn test2(b: bool) -> bool {
13 | |     !b
14 | | }
   | |_^
note: counterexample for "b"
  initial value: true
  final value:   true
  --> $DIR/bool.rs:12:10
   |
12 | fn test2(b: bool) -> bool {
   |          ^
note: counterexample for result
  final value:   false
  --> $DIR/bool.rs:12:22
   |
12 | fn test2(b: bool) -> bool {
   |                      ^^^^

error: [Prusti: verification error] the asserted expression might not hold
  --> $DIR/bool.rs:17:5
   |
17 |     assert!(b);
   |     ^^^^^^^^^^^
   |
note: counterexample for "b"
  initial value: false
  final value:   false
  --> $DIR/bool.rs:16:10
   |
16 | fn test3(b: bool) {
   |          ^
   = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 3 previous errors

